src/Modal.tsx

```diff
@@ -111,6 +112,19 @@ export class ModalManager extends TypedEventEmitter<ModalManagerEvent, HandlerMa
         return container;
     }
 
+    public constructor() {
+        super();
+
+        // We never unregister this: it's a singleton. An instance could be instantiated, in theory...
+        defaultDispatcher.register(this.onAction);
+    }
+
+    private onAction = (payload: ActionPayload): void => {
+        if (payload.action === "logout") {
+            this.closeCurrentModal();
```

Comment:
This won't close all modals given multiple may be stacked.

Reply:
Right, so this is all broken because we use this in several places to close all modals. It won't be a huge PR to fix but it'll be nontrivial: would you prefer it separately or bundled into this one?

Reply:
I think bundled into this one is fine. I can only see one other caller which would benefit from closing all (forced logout case)

Reply:
Pulled out to https://github.com/matrix-org/matrix-react-sdk/pull/12728

Reply:
and now merged